test/built-ins/String/prototype/isWellFormed/returns-boolean.js

```diff
@@ -0,0 +1,20 @@
+// Copyright (C) 2022 Jordan Harband. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-String.prototype.at
+description: >
+  The method should return an Iterator instance.
+info: |
+  String.prototype.isWellFormed ( )
+
+  1. Let O be ? RequireObjectCoercible(this value).
+  2. Let S be ? ToString(O).
+  3. Return IsStringWellFormedUnicode(S).
+
+features: [String.prototype.isWellFormed]
+---*/
+assert.sameValue(typeof String.prototype.isWellFormed, 'function');
+
+assert.sameValue('\uD800'.isWellFormed(), false, 'a string with a lone surrogate is not well-formed');
+assert.sameValue('💩'.isWellFormed(), true, 'a surrogate pair is well-formed');
+assert.sameValue('abc'.isWellFormed(), true, 'a normal string is well-formed');
```

Comment:
The cases I would test:

* a lead surrogate surrounded by other non-surrogates
* a trail surrogate surrounded by other non-surrogates
* adjacent lead/trail surrogates but in the wrong order
* a well-formed string constructed by concatenating a string ending with a lead surrogate and a string starting with a trail surrogate
* a non-well-formed string constructed by slicing a well-formed string in the middle of a surrogate pair
* a string with non-BMP codepoints but no surrogates
* a well-formed string with surrogates
* a latin-1 (7-bit ASCII) string

Reply:
I'm not sure how the string construction method makes a difference?

Reply:
Also, can you suggest some non-surrogate non-BMP codepoints to use?

Reply:
The reason it's important to include tests for a variety of ways that a string can come to be is that it is likely that an implementation trying to provide sub-linear results will need to keep a flag associated with a string. This flag needs to be propagated appropriately across concatenation and slicing. These tests will help ensure that they've done that.

> Also, can you suggest some non-surrogate non-BMP codepoints to use?

Oops, I meant non-ASCII, so anything else in the BMP. Sorry. How about U+25A8 ▨ (SQUARE WITH UPPER RIGHT TO LOWER LEFT FILL)? Also, side note: why do we have a geometric shapes block in the BMP?!